### PR TITLE
fix(parser): improve performance on parsing complex set operations

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes https://github.com/datafuselabs/databend/issues/10802

### CODE REVIEW

File: src/query/ast/tests/it/parser.rs [hunk 1]
```diff
@@ -459,7 +459,7 @@ fn test_query() {
         r#"select * exclude c1, b.* exclude (c2, c3, c4) from customer inner join orders on a = b limit 1"#,
         r#"select * from customer inner join orders"#,
         r#"select * from customer cross join orders"#,
-        r#"select * from customer inner join orders on a = b limit 1"#,
```
Comment (leiysky) [inline src/query/ast/tests/it/parser.rs:462]: Why was this changed?

**Review by leiysky — APPROVED** [review]

File: src/query/ast/tests/it/parser.rs (hunk 1 above)
Comment (andylokandy) [reply src/query/ast/tests/it/parser.rs:462]: To test `ON ~ (#expr)`

File: src/query/ast/src/util.rs [hunk 2]
```diff
@@ -70,7 +70,7 @@ pub fn ident_after_as(i: Input) -> IResult<Identifier> {
 }
 
 pub fn function_name(i: Input) -> IResult<Identifier> {
-    non_reserved_identifier(|token| token.is_reserved_function_name(true))(i)
+    non_reserved_identifier(|token| token.is_reserved_function_name(false))(i)
```
[the remark this answers is not in the document]
Comment (andylokandy) [reply src/query/ast/src/util.rs:73]: RANGE is not reserved in pg, why is it revered in databend?

[the remark this answers is not in the document]
Comment (sundy-li) [reply src/query/ast/src/util.rs]: cause window function need range?
